canopen/objectdictionary/__init__.py

```diff
@@ -15,18 +15,25 @@
 logger = logging.getLogger(__name__)
 
 
-def export_od(od, dest: Union[str, TextIO, None] = None, doc_type: Optional[str] = None):
-    """ Export :class: ObjectDictionary to a file.
+def export_od(
+    od: ObjectDictionary,
+    dest: Union[str, TextIO, None] = None,
+    doc_type: Optional[str] = None
+) -> None:
+    """Export an object dictionary.
 
     :param od:
-        :class: ObjectDictionary object to be exported
+        The object dictionary to be exported.
     :param dest:
-        export destination. filename, or file-like object or None.
-        if None, the document is returned as string
-    :param doc_type: type of document to export.
-       If a filename is given for dest, this default to the file extension.
-       Otherwise, this defaults to "eds"
-    :rtype: str or None
+        The export destination as a filename, a file-like object, or ``None``.
+        If ``None``, the document is written to :data:`sys.stdout`.
+    :param doc_type:
+       The type of document to export.
+       If *dest* is a file-like object or ``None``,
+       *doc_type* must be explicitly provided.
+       If *dest* is a filename and its extension is ``".eds"`` or ``".dcf"``,
+       *doc_type* defaults to that extension (the preceeding dot excluded);
+       else, it defaults to ``"eds"``.
```

Comment:
Just a nit-pick, I find it awkward to have quotes inside quotes here.  Please remove the inner ones.
```suggestion
       If *dest* is a filename and its extension is ``.eds`` or ``.dcf``,
       *doc_type* defaults to that extension (the preceeding dot excluded);
       else, it defaults to ``eds``.
```

Reply:
I'll make the change, though I will point out that it is normal practice to mark up string literals as they are written. Without quotes, there is no visual difference in the docs between identifiers and string literals. See the CPython docs as an example of this convention:

https://docs.python.org/3/library/string.html

Reply:
OK, I didn't understand this as meaning a string literal.  Sure, a variable will hold `".eds"`, but the file's extension is still only `.eds`.  Most of all, I think the code / docstring should be readable (in unrendered form) and the code-style double backquotes are sufficient to see the intention.  It's probably a bit different from CPython docs, where the syntax is more in focus and the docs are hardly ever consumed without rendering.

Reply:
Fair enough!